### PR TITLE
added lesson7

### PR DESCRIPTION
**Pull request by chelseamsmith**

### CODE REVIEW

File: students/ChelseaSmith/lesson7/html_render.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+
+class Element:
+    tag = ""
+    indent = " "
+
+    def __init__(self, content=None, **kwargs):
+        self.content = content
+        self.kwargs = kwargs
+        if self.content != None:
```
Comment (42B) [inline students/ChelseaSmith/lesson7/html_render.py]:
```python
if self.content is not None:  # ?
```

File: students/ChelseaSmith/lesson7/html_render.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        else:
+            self.contentList = []
+
+
+    def append(self, addon):
+        self.contentList.append(addon)
+
+    def render(self, file_out, cur_ind = 0):
+        file_out.write(f"{(self.indent * cur_ind)}<{self.tag}")
+        if type(self.kwargs) == dict:
```
Comment (42B) [inline students/ChelseaSmith/lesson7/html_render.py]:
```python
if isinstance(self.kwargs, dict):  # ?
```

File: students/ChelseaSmith/lesson7/html_render.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            for k in self.kwargs:
+                file_out.write(f" {k}=\"{self.kwargs[k]}\"")   # note: kwargs had 3 empty values in it that prevented me from using for k, v format
+        file_out.write(f">\n")
+        if self.contentList:
+            for item in self.contentList:
+                if type(item) == str:
```
Comment (42B) [inline students/ChelseaSmith/lesson7/html_render.py]:
```python
if isinstance(item, str):  # ?
```

File: students/ChelseaSmith/lesson7/html_render.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+
+class Head(Element):
+    tag = "head"
+
+
+class OneLineTag(Element):
+    def render(self, file_out, cur_ind = 0):
+        file_out.write(f"{(self.indent * cur_ind)}<{self.tag}")
+        if type(self.kwargs) == dict:
```
Comment (42B) [inline students/ChelseaSmith/lesson7/html_render.py]:
```python
if isinstance(self.kwargs, dict):  # ?
```

File: students/ChelseaSmith/lesson7/html_render.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            for k in self.kwargs:
+                file_out.write(
+                    f" {k}=\"{self.kwargs[k]}\"")
+        file_out.write(f">")
+        if self.contentList:
+            for item in self.contentList:
+                if type(item) == str:
```
Comment (42B) [inline students/ChelseaSmith/lesson7/html_render.py]:
```python
if isinstance(item, str):  # ?
```

File: students/ChelseaSmith/lesson7/html_render.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+
+class Title(OneLineTag):
+    tag = "title"
+
+
+class SelfClosingTag(Element):
+    def render(self, file_out, cur_ind = 0):
+        file_out.write(f"{(self.indent * cur_ind)}<{self.tag}")
+        if type(self.kwargs) == dict:
```
Comment (42B) [inline students/ChelseaSmith/lesson7/html_render.py]:
```python
if isinstance(self.kwargs, dict):  # ?
```

File: students/ChelseaSmith/lesson7/html_render.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+class A(Element):
+    tag = "a"
+
+    def __init__(self, link, content):
+        Element.__init__(self, content, href = link)
+
+
+    def render(self, file_out, cur_ind = 0):
+        file_out.write(f"{(self.indent * cur_ind)}<{self.tag}")
+        if type(self.kwargs) == dict:
```
Comment (42B) [inline students/ChelseaSmith/lesson7/html_render.py]:
```python
if isinstance(self.kwargs, dict):  # ?
```

File: students/ChelseaSmith/lesson7/html_render.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+            for k in self.kwargs:
+                file_out.write(
+                    f" {k}=\"{self.kwargs[k]}\"")
+        file_out.write(f">")
+        if self.contentList:
+            for item in self.contentList:
+                if type(item) == str:
```
Comment (42B) [inline students/ChelseaSmith/lesson7/html_render.py]:
```python
if isinstance(item, str):  # ?
```

File: students/ChelseaSmith/lesson7/html_render.py (hunk 1 above)
Comment (natasha-aleksandrova) [inline students/ChelseaSmith/lesson7/html_render.py]: `is` is used with object comparison for `None` / `True` / `False` and other objects.

File: students/ChelseaSmith/lesson7/html_render.py [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    tag = ""
+    indent = " "
+
+    def __init__(self, content=None, **kwargs):
+        self.content = content
+        self.kwargs = kwargs
+        if self.content != None:
+            self.contentList = [content]
+        else:
+            self.contentList = []
```
Comment (natasha-aleksandrova) [inline students/ChelseaSmith/lesson7/html_render.py]: make sure to use snake_case for var names.

File: students/ChelseaSmith/lesson7/test_html_render.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+import html_render
+from io import StringIO
+
+def test_Element_append():
+    sample = html_render.Element("a")
+    sample.append("b")
+    assert sample.contentList == ["a", "b"]
+
+
+def test_Element_render():  # using body subclass as simple content to test render function with
```
Comment (natasha-aleksandrova) [inline students/ChelseaSmith/lesson7/test_html_render.py]: should be snake_case as well

**Pull request merged**